src/main/java/graphql/normalized/VariableAccumulator.java

```diff
@@ -18,15 +19,16 @@
 public class VariableAccumulator {
 
     private final List<VariableValueWithDefinition> valueWithDefinitions;
+    @Nullable
     private final VariablePredicate variablePredicate;
 
-    public VariableAccumulator(VariablePredicate variablePredicate) {
+    public VariableAccumulator(@Nullable VariablePredicate variablePredicate) {
```

Comment:
Was deciding whether to mark the `VariableAccumulator` value as nullable or non-nullable in `compileToDocument` and decided to make it nullable with it defaulting to false.